tests/api/common/experimental/connection_tests.py

```diff
@@ -0,0 +1,427 @@
+            conn_id="new3",
+            conn_uri=uri,
+            conn_extra="{'foo':'bar'}"
+        ), con3_test)
+
+        # add a duplicate connection`
+        self.assertEqual(connections.add_connection(
+            conn_id="new3",
+            conn_uri=uri,
+            conn_extra="{'foo':'bar'}"
+        ), con3_test)
+
+        # add a connection with just type
+        self.assertEqual(connections.add_connection(
+            conn_id="new4",
+            conn_type="google_cloud_platform",
+            conn_uri="",
+            conn_extra="{'extra':'yes'}"
+        ), Connection(conn_id="new4",
+                      conn_type="google_cloud_platform",
+                      uri="",
+                      extra="{'extra':'yes'}"))
+
+        # Attempt to add without providing conn_id
+        with self.assertRaises(MissingArgument) as ma:
+            connections.add_connection(conn_id=None, conn_uri=uri)
+            self.assertEqual(ma.exception,
+                             "The following args are required to add a connection:" +
+                             " ['conn_id']")
+
+        # Attempt to add without providing conn_uri
+        with self.assertRaises(MissingArgument) as ma:
+            connections.add_connection(conn_id="new5")
+            self.assertEqual(ma.exception,
+                             "The following args are required to add a connection:" +
+                             " ['conn_id or conn_type']")
+
+        # Attempt to add duplicate connection with different type
+        with self.assertRaises(IncompatibleArgument) as ia:
+            connections.add_connection(conn_id="new2",
+                                       conn_type="not_postgres",
+                                       conn_host="host",
+                                       conn_login="airflow",
+                                       conn_password="airflow",
+                                       conn_schema="airflow",
+                                       conn_port="5432")
+            self.assertEqual(ia.exception, "Connections with the same id must all be of the same type")
+
+        # validate expected connections reached the DB
+        session = settings.Session()
+        extra = {'new1': None,
+                 'new2': None,
+                 'new3': "{'foo':'bar'}", }
+
+        for index in range(1, 5):
+            conn_id = 'new%s' % index
+            result = (session
+                      .query(models.Connection)
+                      .filter(models.Connection.conn_id == conn_id)
+                      .first())
+            result = (result.conn_id, result.conn_type, result.host,
+                      result.port, result.get_extra())
+            if conn_id in ['new1', 'new2', 'new3']:
+                self.assertEqual(result, (conn_id, 'postgres', 'host', 5432,
+                                          extra[conn_id]))
+            elif conn_id == 'new4':
+                print('result: ', result)
```

Comment:
Generally having bare prints in tests should be avoided - it gets noisy over time